packages/gitbook/src/components/SiteSections/SiteSectionList.tsx

```diff
@@ -239,10 +254,31 @@ function Descendants(props: {
     return (
         <motion.div
             ref={scope}
-            className={isVisible ? undefined : '[&_ul>li]:opacity-1'}
+            className={isVisible ? 'pl-3' : 'pl-3 [&_ul>li]:opacity-1'}
             initial={isVisible ? show : hide}
         >
             {children}
         </motion.div>
     );
 }
+
+/**
+ * Recursively find a section by ID within a group and its nested children
+ */
+function findSectionInGroup(
+    group: ClientSiteSectionGroup,
+    sectionId: string
+): ClientSiteSection | null {
+    for (const child of group.children) {
+        if (child.object === 'site-section' && child.id === sectionId) {
+            return child;
+        }
+        if (child.object === 'site-section-group') {
+            const found = findSectionInGroup(child, sectionId);
+            if (found) {
+                return found;
+            }
+        }
+    }
```

Comment:
Based on my previous suggestion this can be refactored to `filterSectionsFromGroup().find(section => section.id === sectionId)`